product_packaging_type/models/product_packaging_type.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2019 Camptocamp (<http://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+from collections import OrderedDict
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductPackagingType(models.Model):
+    _name = "product.packaging.type"
+    _description = "Type management for product.packaging"
+    _order = "sequence, code"
+
+    name = fields.Char(required=True, translate=True)
+    code = fields.Char(required=True)
+    sequence = fields.Integer(required=True)
+    has_gtin = fields.Boolean()
+    active = fields.Boolean(default=True)
+    is_default = fields.Boolean()
+
+    @api.constrains("is_default")
+    def _check_is_default(self):
+        msg = False
+        default_count = self.search_count([("is_default", "=", True)])
+        if default_count == 0:
+            msg = _('There must be one product packaging type set as "Is Default".')
+        elif default_count > 1:
+            msg = _('Only one product packaging type can be set as "Is Default".')
+        if msg:
+            raise ValidationError(msg)
+
+
+class ProductPackaging(models.Model):
+    _inherit = "product.packaging"
+    _order = "product_id, type_sequence"
```

Comment:
I would order by default based on the contained quantity
```suggestion
    _order = "product_id, qty, type_sequence"
```

Reply:
It makes sense, but then I wonder if the `type_sequence` is really useful then?

Reply:
@jbaudoux ping :)